test/f90_correct/src/io28.f90

```diff
@@ -0,0 +1,38 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! test for I/O take derived types
```

Comment:
Please elaborate on what this test case is checking.

Reply:
> Please elaborate on what this test case is checking.

The comment has been rewritten in the new commit.